# Lecture des fichiers CSV avec Pandas 

In [1]:
import pandas as pd
import numpy as np
#pd.set_option("max_rows", None)  #Préciser le nombre de lignes max

In [2]:
# Lecture des fichiers CSV 
df_population = pd.read_csv("data/fr_population.csv")
df_animaux = pd.read_csv("data/fr_animaux.csv")
df_vegetaux = pd.read_csv("data/fr_vegetaux.csv")
df_cereales = pd.read_csv("data/fr_céréales.csv")
df_sous_alimentation = pd.read_csv("data/fr_sousalimentation.csv")

In [3]:
# Suppression des données dupliquées
df_population.drop_duplicates(keep='first', inplace=True, ignore_index=True)
df_animaux.drop_duplicates(keep='first', inplace=True, ignore_index=True)
df_vegetaux.drop_duplicates(keep='first', inplace=True, ignore_index=True)
df_vegetaux.drop_duplicates(keep='first', inplace=True, ignore_index=True)
df_sous_alimentation.drop_duplicates(keep='first', inplace=True, ignore_index=True)

In [4]:
df_animaux.columns

Index(['Zone', 'Code zone', 'Produit', 'Aliments pour animaux (kg/an)',
       'Autres utilisations (non alimentaire) (kg/an)',
       'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité intérieure (kg/an)', 'Exportations - Quantité (kg/an)',
       'Importations - Quantité (kg/an)', 'Nourriture (kg/an)',
       'Pertes (kg/an)', 'Production (kg/an)', 'Semences (kg/an)',
       'Traitement (kg/an)', 'Variation de stock (kg/an)'],
      dtype='object')

# Aperçu génaral des données

In [5]:
df_population.head(5)

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        202   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          3   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          4   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         79   

             Zone  Code Élément            Élément  Code Produit     Produit  \
0     Afghanistan           511  Population totale          2501  Population   
1  Afrique du Sud           511  Population totale          2501  Population   
2         Albanie           511  Population totale          2501  Population   
3         Algérie           511  Population totale          2501  Population   
4       Allemagne           511  Population totale          2501  Population   

   Code année  Année           Unité  Valeur Symbole Description du Symbole  
0        2013   2013  1000 personnes   30552     NaN      Donnée officielle  
1        2013   2013  1000 personnes   52776     NaN      Donnée officielle  
2        2013   2013  1000 personnes    3173     NaN      Donnée officielle  
3        2013   2013  1000 personnes   39208     NaN      Donnée officielle  
4        2013   2013  1000 personnes   82727     NaN      Donnée officielle

In [6]:
df_population.describe()

Code zone  Code Élément  Code Produit  Code année   Année  \
count  175.000000         175.0         175.0       175.0   175.0   
mean   126.720000         511.0        2501.0      2013.0  2013.0   
std     75.168519           0.0           0.0         0.0     0.0   
min      1.000000         511.0        2501.0      2013.0  2013.0   
25%     64.500000         511.0        2501.0      2013.0  2013.0   
50%    121.000000         511.0        2501.0      2013.0  2013.0   
75%    188.500000         511.0        2501.0      2013.0  2013.0   
max    351.000000         511.0        2501.0      2013.0  2013.0   

             Valeur  
count  1.750000e+02  
mean   4.807996e+04  
std    1.786327e+05  
min    5.400000e+01  
25%    2.543500e+03  
50%    9.413000e+03  
75%    2.888150e+04  
max    1.416667e+06

In [7]:
df_animaux.head(5)

Zone  Code zone                 Produit  \
0  Afghanistan          2        Abats Comestible   
1  Afghanistan          2            Beurre, Ghee   
2  Afghanistan          2                   Crème   
3  Afghanistan          2  Graisses Animales Crue   
4  Afghanistan          2      Lait - Excl Beurre   

   Aliments pour animaux (kg/an)  \
0                            NaN   
1                            NaN   
2                            NaN   
3                            NaN   
4                    123000000.0   

   Autres utilisations (non alimentaire) (kg/an)  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

   Disponibilité alimentaire (Kcal/personne/jour)  \
0                                             5.0   
1                                            23.0   
2                                             3.0   
3                                            12.0   
4                                           100.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.17        
2                                               0.60        
3                                               0.49        
4                                              62.23        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               2.61               
2                                               0.32               
3                                               1.31               
4                                               5.19               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.03          
2                                               0.04          
3                                               0.01          
4                                               6.12          

   Disponibilité intérieure (kg/an)  Exportations - Quantité (kg/an)  \
0                      5.300000e+07                              NaN   
1                      3.600000e+07                              NaN   
2                      1.800000e+07                              NaN   
3                      1.500000e+07                              NaN   
4                      2.083000e+09                              0.0   

   Importations - Quantité (kg/an)  Nourriture (kg/an)  Pertes (kg/an)  \
0                              NaN        5.300000e+07             NaN   
1                              0.0        3.600000e+07             NaN   
2                       18000000.0        1.800000e+07             NaN   
3                        1000000.0        1.500000e+07             NaN   
4                      248000000.0        1.901000e+09      58000000.0   

   Production (kg/an)  Semences (kg/an)  Traitement (kg/an)  \
0        5.300000e+07               NaN                 NaN   
1        3.600000e+07               NaN                 NaN   
2                 NaN               NaN                 NaN   
3        1.400000e+07               NaN                 NaN   
4        1.835000e+09               NaN                 NaN   

   Variation de stock (kg/an)  
0                         NaN  
1                         NaN  
2                         NaN  
3                         NaN  
4                         NaN

In [8]:
df_animaux.describe()

Code zone  Aliments pour animaux (kg/an)  \
count  3731.000000                   5.720000e+02   
mean    126.503082                   2.080839e+08   
std      74.746131                   1.232665e+09   
min       1.000000                   0.000000e+00   
25%      63.000000                   0.000000e+00   
50%     121.000000                   2.000000e+06   
75%     188.500000                   4.025000e+07   
max     351.000000                   2.275700e+10   

       Autres utilisations (non alimentaire) (kg/an)  \
count                                   2.397000e+03   
mean                                    2.116521e+07   
std                                     2.065427e+08   
min                                     0.000000e+00   
25%                                     0.000000e+00   
50%                                     0.000000e+00   
75%                                     0.000000e+00   
max                                     5.972000e+09   

       Disponibilité alimentaire (Kcal/personne/jour)  \
count                                     3726.000000   
mean                                        26.137413   
std                                         58.626273   
min                                          0.000000   
25%                                          0.000000   
50%                                          4.000000   
75%                                         21.750000   
max                                        541.000000   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                        3726.000000        
mean                                            9.300939        
std                                            31.313989        
min                                             0.000000        
25%                                             0.040000        
50%                                             1.050000        
75%                                             5.067500        
max                                           430.760000        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                        3721.000000               
mean                                            1.841387               
std                                             4.192599               
min                                             0.000000               
25%                                             0.000000               
50%                                             0.140000               
75%                                             1.470000               
max                                            41.020000               

       Disponibilité de protéines en quantité (g/personne/jour)  \
count                                        3697.000000          
mean                                            1.753652          
std                                             3.819421          
min                                             0.000000          
25%                                             0.000000          
50%                                             0.210000          
75%                                             1.560000          
max                                            54.970000          

       Disponibilité intérieure (kg/an)  Exportations - Quantité (kg/an)  \
count                      3.731000e+03                     3.021000e+03   
mean                       4.337735e+08                     8.429427e+07   
std                        3.609314e+09                     6.326664e+08   
min                       -9.200000e+07                     0.000000e+00   
25%                        0.000000e+00                     0.000000e+00   
50%                        7.000000e+06                     0.000000e+00   
75%                        6.300000e+07                     8.000000e+06   
max                        1.334430e+11                     1.894

In [9]:
df_vegetaux.head(5)

Zone  Code zone                Produit  \
0  Afghanistan          2        Agrumes, Autres   
1  Afghanistan          2  Aliments pour enfants   
2  Afghanistan          2                 Ananas   
3  Afghanistan          2                Bananes   
4  Afghanistan          2                  Bière   

   Aliments pour animaux (kg/an)  \
0                            NaN   
1                            NaN   
2                            NaN   
3                            NaN   
4                            NaN   

   Autres utilisations (non alimentaire) (kg/an)  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

   Disponibilité alimentaire (Kcal/personne/jour)  \
0                                             1.0   
1                                             1.0   
2                                             0.0   
3                                             4.0   
4                                             0.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.29        
1                                               0.06        
2                                               0.00        
3                                               2.70        
4                                               0.09        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.01               
1                                               0.01               
2                                                NaN               
3                                               0.02               
4                                                NaN               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.02          
1                                               0.03          
2                                                NaN          
3                                               0.05          
4                                               0.00          

   Disponibilité intérieure (kg/an)  Exportations - Quantité (kg/an)  \
0                        41000000.0                        2000000.0   
1                         2000000.0                              NaN   
2                               0.0                              NaN   
3                        82000000.0                              NaN   
4                         3000000.0                              NaN   

   Importations - Quantité (kg/an)  Nourriture (kg/an)  Pertes (kg/an)  \
0                       40000000.0          39000000.0       2000000.0   
1                        2000000.0           2000000.0             NaN   
2                              0.0                 0.0             NaN   
3                       82000000.0          82000000.0             NaN   
4                        3000000.0           3000000.0             NaN   

   Production (kg/an)  Semences (kg/an)  Traitement (kg/an)  \
0           3000000.0               NaN                 NaN   
1                 NaN               NaN                 NaN   
2                 NaN               NaN                 NaN   
3                 NaN               NaN                 NaN   
4                 NaN               NaN                 NaN   

   Variation de stock (kg/an)  
0                         NaN  
1                         NaN  
2                         NaN  
3                         NaN  
4                         NaN

In [10]:
df_vegetaux.describe()

Code zone  Aliments pour animaux (kg/an)  \
count  11971.000000                   2.185000e+03   
mean     126.798847                   7.032165e+08   
std       75.131696                   5.888047e+09   
min        1.000000                   0.000000e+00   
25%       63.000000                   0.000000e+00   
50%      121.000000                   5.000000e+06   
75%      189.000000                   9.400000e+07   
max      351.000000                   1.538020e+11   

       Autres utilisations (non alimentaire) (kg/an)  \
count                                   3.174000e+03   
mean                                    2.820882e+08   
std                                     6.710769e+09   
min                                     0.000000e+00   
25%                                     0.000000e+00   
50%                                     1.000000e+06   
75%                                     1.400000e+07   
max                                     3.473090e+11   

       Disponibilité alimentaire (Kcal/personne/jour)  \
count                                    10607.000000   
mean                                        37.820873   
std                                        119.631129   
min                                        -21.000000   
25%                                          0.000000   
50%                                          4.000000   
75%                                         21.000000   
max                                       1711.000000   

       Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                       10381.000000        
mean                                            8.519772        
std                                            21.822190        
min                                            -1.930000        
25%                                             0.060000        
50%                                             0.770000        
75%                                             5.270000        
max                                           308.070000        

       Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                        8156.000000               
mean                                            1.027108               
std                                             3.395135               
min                                            -0.030000               
25%                                             0.010000               
50%                                             0.070000               
75%                                             0.430000               
max                                            60.760000               

       Disponibilité de protéines en quantité (g/personne/jour)  \
count                                        7944.000000          
mean                                            0.976955          
std                                             3.459216          
min                                            -0.370000          
25%                                             0.010000          
50%                                             0.080000          
75%                                             0.440000          
max                                            49.470000          

       Disponibilité intérieure (kg/an)  Exportations - Quantité (kg/an)  \
count                      1.174700e+04                     9.299000e+03   
mean                       8.768406e+08                     1.224432e+08   
std                        1.169398e+10                     1.157923e+09   
min                       -3.430000e+09                    -4.100000e+07   
25%                        0.000000e+00                     0.000000e+00   
50%                        7.000000e+06                     0.000000e+00   
75%                        8.600000e+07                     9.000000e+06   
max                        7.392670e+11                  

In [11]:
df_cereales.head()

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément     Élément  Code Produit           Produit  \
0  Afghanistan          5511  Production          2511               Blé   
1  Afghanistan          5511  Production          2805  Riz (Eq Blanchi)   
2  Afghanistan          5511  Production          2513              Orge   
3  Afghanistan          5511  Production          2514              Maïs   
4  Afghanistan          5511  Production          2517            Millet   

   Code année  Année               Unité  Valeur Symbole  \
0        2013   2013  Milliers de tonnes    5169       S   
1        2013   2013  Milliers de tonnes     342       S   
2        2013   2013  Milliers de tonnes     514       S   
3        2013   2013  Milliers de tonnes     312       S   
4        2013   2013  Milliers de tonnes      13       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées  
3  Données standardisées  
4  Données standardisées

In [12]:
df_cereales.describe()

Code zone  Code Élément  Code Produit  Code année   Année  \
count  891.000000         891.0    891.000000       891.0   891.0   
mean   130.005612        5511.0   2551.648709      2013.0  2013.0   
std     74.807040           0.0     96.152834         0.0     0.0   
min      1.000000        5511.0   2511.000000      2013.0  2013.0   
25%     67.000000        5511.0   2514.000000      2013.0  2013.0   
50%    129.000000        5511.0   2516.000000      2013.0  2013.0   
75%    197.500000        5511.0   2518.000000      2013.0  2013.0   
max    276.000000        5511.0   2805.000000      2013.0  2013.0   

              Valeur  
count     891.000000  
mean     2832.037037  
std     16786.590126  
min         0.000000  
25%         7.000000  
50%        95.000000  
75%       824.000000  
max    353699.000000

In [13]:
df_sous_alimentation.head()

Code Domaine                             Domaine  Code zone         Zone  \
0           FS  Données de la sécurité alimentaire          2  Afghanistan   
1           FS  Données de la sécurité alimentaire          2  Afghanistan   
2           FS  Données de la sécurité alimentaire          2  Afghanistan   
3           FS  Données de la sécurité alimentaire          2  Afghanistan   
4           FS  Données de la sécurité alimentaire          2  Afghanistan   

   Code Élément Élément  Code Produit  \
0          6132  Valeur        210011   
1          6132  Valeur        210011   
2          6132  Valeur        210011   
3          6132  Valeur        210011   
4          6132  Valeur        210011   

                                             Produit  Code année      Année  \
0  Nombre de personnes sous-alimentées (millions)...    20122014  2012-2014   
1  Nombre de personnes sous-alimentées (millions)...    20132015  2013-2015   
2  Nombre de personnes sous-alimentées (millions)...    20142016  2014-2016   
3  Nombre de personnes sous-alimentées (millions)...    20152017  2015-2017   
4  Nombre de personnes sous-alimentées (millions)...    20162018  2016-2018   

      Unité Valeur Symbole Description du Symbole  Note  
0  millions    7.9       F         Estimation FAO   NaN  
1  millions    8.8       F         Estimation FAO   NaN  
2  millions    9.6       F         Estimation FAO   NaN  
3  millions   10.2       F         Estimation FAO   NaN  
4  millions   10.6       F         Estimation FAO   NaN

In [14]:
df_sous_alimentation.describe()

Code zone  Code Élément  Code Produit    Code année  Note
count  1020.000000        1020.0        1020.0  1.020000e+03   0.0
mean    129.264706        6132.0      210011.0  2.014202e+07   NaN
std      76.615019           0.0           0.0  1.415049e+04   NaN
min       1.000000        6132.0      210011.0  2.012201e+07   NaN
25%      62.500000        6132.0      210011.0  2.013202e+07   NaN
50%     127.500000        6132.0      210011.0  2.014202e+07   NaN
75%     193.250000        6132.0      210011.0  2.015202e+07   NaN
max     351.000000        6132.0      210011.0  2.016202e+07   NaN

# Réponse 1  : Nombre total d’humains sur la planète

In [15]:
df_population2013 = df_population[df_population['Code année']==2013].copy()
df_population2013['Valeur'].sum() * 1000
#L'unité utilisé dans notre db est 1000 personnes

8413993000

## Critique du résultat obtenu

<ul>
    <li> Pour le nombre total d'humains en 2013, nous avons obtenue : <b>8 413 993 000</b> personnes </li>
    <li>Or, selon <a href = https://www.worldbank.org/>la banque mondiale</a> la population mondiale en 2013 est environ 7 Md d'humaines </li>
    <li>Afin de clarifier notre résultat nous allons :      
        <ol>            
            <li>Vérifier si le nombre de population pour un pays donnée est calculé plusieurs fois</li>
            <li>Les valeurs manquantes pour année ou code année</li>
            <li>Observer les données <a href = https://www.fao.org/faostat/fr/#definitions>non officielles</a>  </li>                </ol>
    </li> 
</ul>

In [16]:
df_population2013['Zone'].value_counts()

Afghanistan    1
Oman           1
Namibie        1
Népal          1
Nicaragua      1
              ..
France         1
Gabon          1
Gambie         1
Géorgie        1
Zimbabwe       1
Name: Zone, Length: 175, dtype: int64

In [17]:
df_population2013['Code année'].isna().sum(),df_population2013['Année'].isna().sum()

(0, 0)

In [18]:
df_population2013[~df_population2013['Symbole'].isna()]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [19]:
df_Chine_A = df_population[df_population['Zone'].str.match(pat = '(^Chin.)')] 
df_Chine_A

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   
34         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         96   
35         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        128   
36         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         41   
37         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        214   

                         Zone  Code Élément            Élément  Code Produit  \
33                      Chine           511  Population totale          2501   
34   Chine - RAS de Hong-Kong           511  Population totale          2501   
35       Chine - RAS de Macao           511  Population totale          2501   
36        Chine, continentale           511  Population totale          2501   
37  Chine, Taiwan Province de           511  Population totale          2501   

       Produit  Code année  Année           Unité   Valeur Symbole  \
33  Population        2013   2013  1000 personnes  1416667       A   
34  Population        2013   2013  1000 personnes     7204     NaN   
35  Population        2013   2013  1000 personnes      566     NaN   
36  Population        2013   2013  1000 personnes  1385567     NaN   
37  Population        2013   2013  1000 personnes    23330     NaN   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...  
34                                  Donnée officielle  
35                                  Donnée officielle  
36                                  Donnée officielle  
37                                  Donnée officielle

In [20]:
df_Chine_A[df_Chine_A['Zone']!='Chine']['Valeur'].sum()

1416667

Nous remarquons que la population la zone chine est une agrégation de plusieurs autres zones, si nous calculons la somme de population de ces dernières, nous retouvons qu'elle est égale à la population chinoise, c'est pourquoi nous allons calculer la population mondiale sans prendre en considération la chine

## Suppression des agrégats de la chine dans toutes les dataframes 

In [21]:
df_population = df_population2013[~df_population2013['Code zone'].isin([41, 96,128,214])].copy()
df_animaux = df_animaux[~df_animaux['Code zone'].isin([41, 96,128,214])].copy()
df_vegetaux = df_vegetaux[~df_vegetaux['Code zone'].isin([41, 96,128,214])].copy()
df_cereales = df_cereales[~df_cereales['Code zone'].isin([41, 96,128,214])].copy()
df_sous_alimentation = df_sous_alimentation[~df_sous_alimentation['Code zone'].isin([41, 96,128,214])].copy()

<b>Ou bien Suppression de la chine des autres bases de données </b>

<p>df_animaux = df_animaux[df_animaux['Zone']!='Chine'].copy() <br>
df_vegetaux = df_vegetaux[df_vegetaux['Zone']!='Chine'].copy()<br>
df_cereales = df_cereales[df_cereales['Zone']!='Chine'].copy()<br>
df_sous_alimentation = df_sous_alimentation[df_sous_alimentation['Zone']!='Chine'].copy()</p>


In [22]:
#L'unité utilisé dans notre db est 1000 personnes
nb_population = df_population['Valeur'].sum() * 1000
print(nb_population)

6997326000


============================================================== <br>
<B><I>Le nombre total d'humain en 2013 est 6997326000 </I></B> <br> ==============================================================

# Réponse 2 : Equation à 3 termes

======================================================================= <br>
<b><i>Disponibilité intérieure = <br>Production + Importations + |Variation de stock| - Exportations = 
<br>Aliments pour animaux + Semences + Pertes + Traitement + Autres utilisations (non alimentaire) + Nourriture
 </i></b>
======================================================================= <br>


<strong><u> Exemple blé en france 2018 </u></strong>

In [23]:
# Données récupérées à partir du lien : https://www.fao.org/faostat/fr/#data/FBS
data_ble_france2018= pd.read_csv('data/ble_france_2018.csv')
data_ble_france2018.head(3)

Code Domaine                      Domaine  Code zone (FAO)    Zone  \
0          FBS  Bilans Alimentaires (2014-)               68  France   
1          FBS  Bilans Alimentaires (2014-)               68  France   
2          FBS  Bilans Alimentaires (2014-)               68  France   

   Code Élément                  Élément  Code Produit          Produit  \
0          5511               Production          2511  Blé et produits   
1          5611  Importations - Quantité          2511  Blé et produits   
2          5072       Variation de stock          2511  Blé et produits   

   Code année  Année               Unité  Valeur Symbole  \
0        2018   2018  Milliers de tonnes   35798      Im   
1        2018   2018  Milliers de tonnes    2349      Im   
2        2018   2018  Milliers de tonnes   -1819      Im   

                              Description du Symbole  
0  Données de la FAO basées sur une méthodologie ...  
1  Données de la FAO basées sur une méthodologie ...  
2  Données de la FAO basées sur une méthodologie ...

In [24]:
data_ble_france2018 = data_ble_france2018[['Code Élément', 'Élément', 'Unité', 'Valeur']].copy()

In [25]:
#Le premier terme
terme1=data_ble_france2018[data_ble_france2018['Code Élément']==5301]['Valeur'].iloc[0] #valeur de disponibilité intérieur

In [26]:
#Le second terme
terme2=data_ble_france2018[data_ble_france2018['Code Élément']==5511]['Valeur'].iloc[0] + \
data_ble_france2018[data_ble_france2018['Code Élément']==5611]['Valeur'].iloc[0] + \
data_ble_france2018[data_ble_france2018['Code Élément']==5072]['Valeur'].abs() .iloc[0]- \
data_ble_france2018[data_ble_france2018['Code Élément']==5911]['Valeur'].iloc[0]

In [27]:
#Le troisième terme
terme3=data_ble_france2018[data_ble_france2018['Code Élément']==5521]['Valeur'].iloc[0] +\
    data_ble_france2018[data_ble_france2018['Code Élément']==5527]['Valeur'].iloc[0] + \
    data_ble_france2018[data_ble_france2018['Code Élément']==5123]['Valeur'].iloc[0]+ \
    data_ble_france2018[data_ble_france2018['Code Élément']==5131]['Valeur'].iloc[0] + \
    data_ble_france2018[data_ble_france2018['Code Élément']==5154]['Valeur'].iloc[0]+ \
    data_ble_france2018[data_ble_france2018['Code Élément']==5142]['Valeur'].iloc[0]

In [28]:
print('Terme1 = ',terme1)
print('Terme2 = ',terme2)
print('Terme2 = ',terme3)

Terme1 =  19521
Terme2 =  19520
Terme2 =  19521


 <b>Exemple blé en france en 2013 (à partir de notre db): voir projet3 archivé </b>

# Réponse 3 : Calculer pour chaque pays et chaque produit la disponibilité alimentaire en kcal puis en kg de protéines.

## Etape 1  : garder uniquement les colonnes importantes et rajouter une colonne "type" pour distinguer entre df_animaux et df_vegetaux

In [29]:
df_population1 = df_population[['Code zone','Valeur','Unité']].copy()

In [30]:
df_animaux1 = df_animaux.assign(Type = 'animal')
df_vegetaux1 = df_vegetaux.assign(Type = 'vegetal')

## Etape 2  : concaténation entre df_animaux1 et df_vegetaux1

In [31]:
df_Alimentation1=pd.concat([df_animaux1,df_vegetaux1])

In [32]:
print('df_animaux1.shape =\t', df_animaux1.shape , '\ndf_vegetaux2.shape =\t',
      df_vegetaux1.shape,'\ndf_Alimentation1.shape =\t',df_Alimentation1.shape)

df_animaux1.shape =	 (3643, 19) 
df_vegetaux2.shape =	 (11682, 19) 
df_Alimentation1.shape =	 (15325, 19)


## Etape3 : Jointure à gauche entre df_Alimentation1, df_population1 

In [33]:
df_prinicpale = df_Alimentation1.merge( df_population1, on='Code zone', how="left", suffixes=('_produit','_population'))
df_prinicpale.rename(columns={'Valeur' : 'Population'}, inplace=True)

In [34]:
print("df_Alimentation1.shape =\t",df_Alimentation1.shape,'\ndf_population1.shape =\t',df_population1.shape,
      '\ndf_alimentation_population.shape =',df_prinicpale.shape)
df_prinicpale.head(3)

df_Alimentation1.shape =	 (15325, 19) 
df_population1.shape =	 (171, 3) 
df_alimentation_population.shape = (15325, 21)


Zone  Code zone           Produit  Aliments pour animaux (kg/an)  \
0  Afghanistan          2  Abats Comestible                            NaN   
1  Afghanistan          2      Beurre, Ghee                            NaN   
2  Afghanistan          2             Crème                            NaN   

   Autres utilisations (non alimentaire) (kg/an)  \
0                                            NaN   
1                                            NaN   
2                                            NaN   

   Disponibilité alimentaire (Kcal/personne/jour)  \
0                                             5.0   
1                                            23.0   
2                                             3.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.17        
2                                               0.60        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               2.61               
2                                               0.32               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.03          
2                                               0.04          

   Disponibilité intérieure (kg/an)  ...  Importations - Quantité (kg/an)  \
0                        53000000.0  ...                              NaN   
1                        36000000.0  ...                              0.0   
2                        18000000.0  ...                       18000000.0   

   Nourriture (kg/an)  Pertes (kg/an)  Production (kg/an)  Semences (kg/an)  \
0          53000000.0             NaN          53000000.0               NaN   
1          36000000.0             NaN          36000000.0               NaN   
2          18000000.0             NaN                 NaN               NaN   

   Traitement (kg/an)  Variation de stock (kg/an)    Type Population  \
0                 NaN                         NaN  animal      30552   
1                 NaN                         NaN  animal      30552   
2                 NaN                         NaN  animal      30552   

            Unité  
0  1000 personnes  
1  1000 personnes  
2  1000 personnes  

[3 rows x 21 columns]

## Etape 4 : Calcul de disponibilité alimentaire en kcal puis en kg 

### 1. Disponibilité alimentaire en kcal 
<ul>
    <li>Disponibilite alimenatire en kcal = <br>'Disponibilité alimentaire (Kcal/personne/jour)' * Nombre d'habitants * unité (1000) * 365 jours</li>
</ul>

In [35]:
df_prinicpale['Disponibilite alimentaire (kcal/an)'] = df_prinicpale['Disponibilité alimentaire (Kcal/personne/jour)'] * 365 * df_prinicpale['Population'] * 1000

### 2. Disponibilité alimentaire en kg de protéine 
<ul>
    <li>Disponibilite alimenatire en kg de proteine = <br>Disponibilité de protéines en quantité (g/personne/jour) / 1000 (g --> kg) * Nombre d'habitants * unité (1000) * 365 jours</li>
</ul>
Nb : Pour optimiser le calcul, nous n'allons pas multiplier par 1000 puis diviser par 1000

In [36]:
df_prinicpale['Disponibilite alimentaire proteine (kg/an)'] = df_prinicpale['Disponibilité de protéines en quantité (g/personne/jour)'] * 365 * df_prinicpale['Population'] 
#Affichage du Df
df_prinicpale[['Zone','Produit','Disponibilite alimentaire (kcal/an)','Disponibilite alimentaire proteine (kg/an)']]

Zone                 Produit  \
0       Afghanistan        Abats Comestible   
1       Afghanistan            Beurre, Ghee   
2       Afghanistan                   Crème   
3       Afghanistan  Graisses Animales Crue   
4       Afghanistan      Lait - Excl Beurre   
...             ...                     ...   
15320  Îles Salomon                  Sésame   
15321  Îles Salomon                     Thé   
15322  Îles Salomon                 Tomates   
15323  Îles Salomon                     Vin   
15324  Îles Salomon          Épices, Autres   

       Disponibilite alimentaire (kcal/an)  \
0                             5.575740e+10   
1                             2.564840e+11   
2                             3.345444e+10   
3                             1.338178e+11   
4                             1.115148e+12   
...                                    ...   
15320                         0.000000e+00   
15321                         0.000000e+00   
15322                         0.000000e+00   
15323                         0.000000e+00   
15324                         8.190600e+08   

       Disponibilite alimentaire proteine (kg/an)  
0                                      8586639.60  
1                                       334544.40  
2                                       446059.20  
3                                       111514.80  
4                                     68247057.60  
...                                           ...  
15320                                         NaN  
15321                                     8190.60  
15322                                         NaN  
15323                                         NaN  
15324                                    30714.75  

[15325 rows x 4 columns]

# Réponse 4 

## 1. Ratio "énergie/poids" (en kcal/kg) pour chaque produit et chaque pays

In [37]:
#Projetction de df_principale
df_ratio = df_prinicpale[['Zone','Produit','Nourriture (kg/an)','Disponibilite alimentaire (kcal/an)','Disponibilite alimentaire proteine (kg/an)']].copy()

In [38]:
# Remplacer les valeurs égales à 0 par des valeurs manquantes pour éviter la division sur 0
df_ratio.replace(0, np.nan, inplace=True)

In [39]:
df_ratio['Ratio Energie/poids (kcal/kg)'] = \
df_ratio['Disponibilite alimentaire (kcal/an)']/  df_ratio["Nourriture (kg/an)"]

## 2. Ratio "poids de protéines/poids total" (en g/kg) pour chaque produit et chaque pays

In [40]:
df_ratio['Ratio protéine/poids'] = df_ratio['Disponibilite alimentaire proteine (kg/an)']/ \
df_ratio["Nourriture (kg/an)"]

In [41]:
df_ratio = df_ratio.loc[:,['Produit','Ratio Energie/poids (kcal/kg)','Ratio protéine/poids']].copy()

In [42]:
df_ratio

Produit  Ratio Energie/poids (kcal/kg)  \
0            Abats Comestible                    1052.026415   
1                Beurre, Ghee                    7124.556667   
2                       Crème                    1858.580000   
3      Graisses Animales Crue                    8921.184000   
4          Lait - Excl Beurre                     586.611257   
...                       ...                            ...   
15320                  Sésame                            NaN   
15321                     Thé                            NaN   
15322                 Tomates                            NaN   
15323                     Vin                            NaN   
15324          Épices, Autres                            NaN   

       Ratio protéine/poids  
0                  0.162012  
1                  0.009293  
2                  0.024781  
3                  0.007434  
4                  0.035901  
...                     ...  
15320                   NaN  
15321                   NaN  
15322                   NaN  
15323                   NaN  
15324                   NaN  

[15325 rows x 3 columns]

<b> Moyenne du ratio pour chaque aliment </b> 

In [43]:
df_ratio = df_ratio.groupby('Produit').mean().reset_index()

## Exemples  
L'apport énergitique <b> des oeufs crus</b> est de 147 kcal pour 100g [<a href='https://fr.wikipedia.org/wiki/%C5%92uf_(aliment)'> Wikipedia </a>], soit 1470 par kg et nous avons trouvé un ratio moyen de 1343,44 kcal/kg.

In [44]:
df_ratio[df_ratio['Produit']=='Oeufs']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
58   Oeufs                    1343.440349              0.105473

<b> L'avoine</b> contient 10,70 g de proteine pour 100g <a href ='https://fr.wikipedia.org/wiki/Avoine_cultiv%C3%A9e'> Wikipedia </a>, soit un ratio de 0,107, et nous avons trouvé un ratio moyen de 0.084.

In [45]:
df_ratio[df_ratio['Produit']=='Avoine']

Produit  Ratio Energie/poids (kcal/kg)  Ratio protéine/poids
7  Avoine                     2135.35579              0.084294

# Réponse 5 

## 1. Les 5 aliments les plus calorique 

In [46]:
Elts_caloriques = df_ratio.sort_values(by='Ratio Energie/poids (kcal/kg)', ascending=False).head(5)
Elts_caloriques[['Produit','Ratio Energie/poids (kcal/kg)']]

Produit  Ratio Energie/poids (kcal/kg)
45  Huiles de Foie de Poisso                    9982.932500
42       Huile de Son de Riz                    9379.568137
43           Huile de Sésame                    9314.546692
46        Huiles de Poissons                    9003.715504
34          Huile d'Arachide                    8925.582634

## 2. Les 5 aliments les plus protéinés</b>

In [47]:
Elts_proteinees = df_ratio.sort_values(by='Ratio protéine/poids', ascending=False).head(5)
Elts_proteinees[['Produit','Ratio protéine/poids']]


Produit  Ratio protéine/poids
27  Graines Colza/Moutarde              0.285365
81                    Soja              0.285156
6   Arachides Decortiquees              0.251573
70                    Pois              0.223219
50     Légumineuses Autres              0.220334

<b> Rajout dans la df principale les ratios </b>

In [48]:
df_prinicpale = df_prinicpale.merge(df_ratio, on='Produit').copy()

#  Réponse 6 : Disponibilité intérieure mondiale pour les produits végétaux 

In [49]:
df_alimentation_vegetal = df_prinicpale[df_prinicpale['Type'] == 'vegetal'].copy()

## 1. Disponibilité intérieure mondiale en kcal (pour les produits végétaux uniquement)

In [50]:
df_alimentation_vegetal['Disponibilité intérieure (kcal/an)'] = df_alimentation_vegetal["Disponibilité intérieure (kg/an)"] * \
    df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"] 
total_disponibilite_interieur_vegetal_kcal = df_alimentation_vegetal['Disponibilité intérieure (kcal/an)'].sum()
print('Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement = ',total_disponibilite_interieur_vegetal_kcal)

Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement =  1.3780136677764188e+16


================================================================================<br>
Disponibilité intérieure mondiale en kcal pour les produits végétaux uniquement =  <b>1.3780136677764184e+16</b> <br>
================================================================================

## 2. Disponibilité intérieure mondiale en proteine (pour les produits végétaux uniquement)

In [51]:
df_alimentation_vegetal['Disponibilité intérieure en proteine (g/an)'] = df_alimentation_vegetal["Disponibilité intérieure (kg/an)"] * \
    df_alimentation_vegetal["Ratio protéine/poids"] * 1000
total_disponibilite_interieur_vegetal_proteine = df_alimentation_vegetal['Disponibilité intérieure en proteine (g/an)'].sum()
print('total_disponibilite_interieur_vegetal_proteine = ',total_disponibilite_interieur_vegetal_proteine)

total_disponibilite_interieur_vegetal_proteine =  349177827738287.5


================================================================================ <br>
Disponibilité intérieure mondiale en proteine pour les produits végétaux uniquement =  <b>349177827738287.5 g</b> <br>
================================================================================

# Réponse 7 : Nombre d'humains qui pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture 

## 1. Selon le nombre de calories

In [52]:
nombre_personne_nourris_disponibilite_interieur_vegetal_kcal =total_disponibilite_interieur_vegetal_kcal / (2500 * 365)
pourcentage_nourris_disponibilite_interieur_vegetal_kcal = nombre_personne_nourris_disponibilite_interieur_vegetal_kcal *100/nb_population
print ('Nombre de personnes nourries =', nombre_personne_nourris_disponibilite_interieur_vegetal_kcal)
print('Pourcentage des personnes nourries =', pourcentage_nourris_disponibilite_interieur_vegetal_kcal,'%')

Nombre de personnes nourries = 15101519646.864864
Pourcentage des personnes nourries = 215.8184375983749 %


Sachant que la ration alimentaire moyenne nécessaire est de <b>2 500 kcal/personne/jour</b>. [Source <a href="https://fr.wikipedia.org/wiki/Ration_alimentaire#cite_note-2">Wikipedia</a> ] , alors :<br>
Le nombre de personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = <b>15101519646</b><br>
Pourcentage des personnes nourries en utilisant la disponibilite interieur vegetal selon de nombre de calories = <b>215.89 %</b>

## 2. Selon le nombre de proteines 

In [53]:
nombre_personne_nourris_disponibilite_interieur_vegetal_proteine =total_disponibilite_interieur_vegetal_proteine / (58.5 * 365)
pourcentage_nourris_disponibilite_interieur_vegetal_proteine = nombre_personne_nourris_disponibilite_interieur_vegetal_proteine *100/nb_population
print ('Nombre de personnes nourries =', nombre_personne_nourris_disponibilite_interieur_vegetal_proteine)
print('pourcentage de personnes nourries =', pourcentage_nourris_disponibilite_interieur_vegetal_proteine,'%')

Nombre de personnes nourries = 16353018510.164501
pourcentage de personnes nourries = 233.7038250063596 %


Le besoin moyen de protéine par kg est de 0.9g, sachant le poids moyen d'une personne est de 65 kg, alors le besoin moyen en protéine est de <b>58.5 g/personne </b>  
Le nombre de personne nourris en utilisant la disponibilite interieur vegetal en terme de proteine = <b>16353018510</b> <br>
Le pourcentage = <b>233.7 %</b>

# Réponse 8 : Nombre d'humains nourris si toute la disponibilité alimentaire en produits végétaux, la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture 

## 1. Selon le nombre de calories :

In [54]:
# df_alimentation_vegetal['Disponibilite sans perte (kcal)'] = 
# (df_alimentation_vegetal['Aliments pour animaux (kg/an)']+\
#                       df_alimentation_vegetal['Pertes (kg/an)'] + \
#                      df_alimentation_vegetal['Nourriture (kg/an)'])   \
#                  * df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"] 
 
# total_disponibilite_sans_perte_vegetal_kcal = df_alimentation_vegetal['Disponibilite sans perte (kcal)'].sum()
# print("total_disponibilite_sans_perte_vegetal_kcal = ",total_disponibilite_sans_perte_vegetal_kcal)

In [55]:
# b = df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"] .sum()
# b *df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"]

In [56]:
# ((df_alimentation_vegetal['Aliments pour animaux (kg/an)']+\
#                       df_alimentation_vegetal['Pertes (kg/an)'] + \
#                      df_alimentation_vegetal['Nourriture (kg/an)']).sum() * df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"]).sum()

In [57]:
df_alimentation_vegetal['Disponibilite sans perte'] = \
df_alimentation_vegetal[['Aliments pour animaux (kg/an)','Pertes (kg/an)','Nourriture (kg/an)']].sum(axis=1) 
total_disponibilite_sans_perte_vegetal_kcal = (df_alimentation_vegetal['Disponibilite sans perte'] * df_alimentation_vegetal["Ratio Energie/poids (kcal/kg)"] ).sum()

In [58]:
nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal =total_disponibilite_sans_perte_vegetal_kcal / (2500 * 365)
pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal = nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal *100/nb_population
print ('nombre_personne_nourris_kcal =', nombre_personne_nourris_disponibilite_sans_perte_vegetal_kcal)
print('pourcentage_nourris_kcal =', pourcentage_nourris_disponibilite_sans_perte_vegetal_kcal,'%')

nombre_personne_nourris_kcal = 10415832398.520512
pourcentage_nourris_kcal = 148.85446809996435 %


Nombre de personnes nourris si la toute la disponibilite alimentaire vegétale est utilisé (sans perte ou nourriture pour animaux) selon le besoin moyen en kcal = <b>10415832398</b><br>
Pourcentage des personnes nourris = <b>148.85 %</b> 

## 2. Selon le nombre de proteines

In [59]:
# perte_animaux_proteine = (df_alimentation_vegetal['Aliments pour animaux (kg/an)']+df_alimentation_vegetal.loc[:,'Pertes (kg/an)']) \
#                  * df_alimentation_vegetal.loc[:,"Ratio protéine/poids"] 
# df_alimentation_vegetal.loc[:,'Disponibilite sans perte proteine(kg/an)'] = perte_animaux_proteine + df_alimentation_vegetal.loc[:,'Disponibilite alimentaire proteine (kg/an)'] 
# total_disponibilite_sans_perte_vegetal_proteine = df_alimentation_vegetal['Disponibilite sans perte proteine(kg/an)'].sum() * 1000
# print("total_disponibilite_sans_perte_vegetal_proteine en g= ",total_disponibilite_sans_perte_vegetal_proteine)

In [60]:
total_disponibilite_sans_perte_vegetal_proteine = (df_alimentation_vegetal['Disponibilite sans perte'] * df_alimentation_vegetal["Ratio protéine/poids"]).sum()
# La disponibilité en gramme = 
total_disponibilite_sans_perte_vegetal_proteine *=1000

In [61]:
nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine =total_disponibilite_sans_perte_vegetal_proteine / (58.5 * 365)
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine *100/nb_population
print ('nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine =', nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine)
print('pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine =', pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine,'%')

nombre_personne_nourris_disponibilite_sans_perte_vegetal_proteine = 10047984897.607777
pourcentage_nourris_disponibilite_sans_perte_vegetal_proteine = 143.59749563773042 %


Nombre de personnes nourris si la toute la disponibilite alimentaire vegétale est utilisé (sans perte ou nourriture pour animaux) selon le besoin moyen en proteine = <b>10047984897</b><br>
Pourcentage des personnes nourris = <b>143.6 %</b>

# Réponse 9 : Nombre d'humains qui pourraient être nourris avec la disponibilité alimentaire mondiale

## 1. Selon le nombre de kca

In [62]:
disponibilite_globale_kcal = df_prinicpale['Disponibilite alimentaire (kcal/an)'].sum()

In [63]:
nb_personne_nourris_disponibilite_gobale_kcal =disponibilite_globale_kcal / (2500 * 365)
pourcentage_personne_nourris_disponibilite_gobale_kcal = nb_personne_nourris_disponibilite_gobale_kcal *100 / nb_population
print('disponibilite_globale_kcal = ',disponibilite_globale_kcal)
print ('nb_personne_nourris_disponibilite_gobale_kcal =', nb_personne_nourris_disponibilite_gobale_kcal)
print('pourcentage_personne_nourris_disponibilite_gobale_kcal =', pourcentage_personne_nourris_disponibilite_gobale_kcal,'%')

disponibilite_globale_kcal =  7364437353675000.0
nb_personne_nourris_disponibilite_gobale_kcal = 8070616278.0
pourcentage_personne_nourris_disponibilite_gobale_kcal = 115.33857759378368 %


Disponibilité globale en kcal =  <b>7364437353675000</b><br>
Nombre de personnes qui peuvent être nourris avec la disponibilite gobale selon leur besoin en kcal = <b>8070616278.0</b><br>
pourcentage par rapport à la population globale = <b>115.33857759378368 %</b><br>

## 2. Selon le nombre de proteines 

In [64]:
disponibilite_globale_proteine = df_prinicpale['Disponibilite alimentaire proteine (kg/an)'].sum() * 1000

In [65]:
nb_personne_nourris_disponibilite_gobale_proteine =disponibilite_globale_proteine / (58.5 * 365)
pourcentage_personne_nourris_disponibilite_gobale_proteine = nb_personne_nourris_disponibilite_gobale_proteine *100 / nb_population
print('disponibilite_globale_proteine = ',disponibilite_globale_proteine)
print ('nb_personne_nourris_disponibilite_gobale_proteine =', nb_personne_nourris_disponibilite_gobale_proteine)
print('pourcentage_personne_nourris_disponibilite_gobale_proteine =', pourcentage_personne_nourris_disponibilite_gobale_proteine,'%')

disponibilite_globale_proteine =  207249144517349.97
nb_personne_nourris_disponibilite_gobale_proteine = 9706083339.999998
pourcentage_personne_nourris_disponibilite_gobale_proteine = 138.71132115325193 %


Disponibilite globale en proteine =  <b>207249144517349.97</b><br>
Nombre de personnes qui peuvent être nourris avec la disponibilite gobale selon leur besoin en proteine  = <b>9 706 083 340</b><br>
pourcentage par rapport à la population globale = <b>138.71 %</b>


# Réponse 10 : Proportion de la population mondiale considérée comme étant en sous-nutrition


In [66]:
# df_sous_alimentation2013 = df_sous_alimentation[df_sous_alimentation['Code année']==20122014].copy()
df_sous_alimentation2013 = df_sous_alimentation[df_sous_alimentation['Année'].str[:4]=='2012'].copy()
# Remplacer l'année et le code année qui commencent par 2012 par 2013 
df_sous_alimentation2013[['Code année','Année']] = 2013
# Projection de la df_sous_alimentation2013
df_sous_alimentation2013.drop(columns=['Code Domaine', 'Domaine','Code Élément','Élément', 'Code Produit', 'Produit', 'Symbole', 'Description du Symbole', 'Note'] , inplace=True)

In [67]:
# df_sous_alimentation2013['Valeur'] est de type Objet, on va le caster en type float
df_sous_alimentation2013['Valeur'] = pd.to_numeric(df_sous_alimentation2013['Valeur'], errors = "coerce")
# En utilisant la fonction to_numeric, les valeurs de type String ('<0.1') vont devenir Null
nb_sous_alimentation2013 = df_sous_alimentation2013['Valeur'].sum() * 1000000
print('nb_sous_alimentation2013 = ',nb_sous_alimentation2013)
pourcentage_sous_alimentation2013  = nb_sous_alimentation2013 *100/ nb_population

print('pourcentage_sous_alimentation2013 = ',pourcentage_sous_alimentation2013)

nb_sous_alimentation2013 =  743800000.0000001
pourcentage_sous_alimentation2013 =  10.629774859710697


Nombre de personnes considérées comme étant en sous-nutrition : <b>743800000</b><br>
proportion de la population mondiale considérée comme étant en sous-nutrition : <b>10.63</b>

In [68]:
## df_sous_alimentation2013[df_sous_alimentation2013['Zone']=="Dominique"]

# Réponse 11 : Proportion destinée à l'alimentation animale en ne prenant en compte que les céréales destinées à l'alimentation 

<b> a. Liste des produits considérés comme des céréales selon la FAO.</b>

In [69]:
type_cereal = df_cereales['Produit'].unique()
type_cereal

array(['Blé', 'Riz (Eq Blanchi)', 'Orge', 'Maïs', 'Millet', 'Seigle',
       'Avoine', 'Sorgho', 'Céréales, Autres'], dtype=object)

<b> b. Rajout de la colonne "is_cereal"de type booléen pour distinguer les céréales des autres produits </b>

In [70]:
df_prinicpale['is_cereal'] = df_prinicpale['Produit'].isin(type_cereal)

In [71]:
# Ou bien
# df_type_cereal = df_cereales.drop_duplicates(subset = ['Code Produit', 'Produit'])
# type_cereal = df_type_cereal["Produit"].tolist()
# df_prinicpale['is_cereal'] = df_prinicpale['Produit'].apply(lambda x:True if x in type_cereal else False)

In [72]:
df_prinicpale_cereal = df_prinicpale[df_prinicpale['is_cereal']].copy() # Masque is_cereal = True
#df_prinicpale_cereal[['Zone','Produit','Aliments pour animaux (kg/an)', 'Nourriture (kg/an)']]

In [73]:
cereales_animaux = df_prinicpale_cereal['Aliments pour animaux (kg/an)'].sum()
cereales_nourriture = df_prinicpale_cereal['Nourriture (kg/an)'].sum()

proportion_cereales_animaux = 100 * cereales_animaux / (cereales_animaux + cereales_nourriture)
#proportion_cereales_nourriture = 100 * cereales_nourriture / (cereales_animaux + cereales_nourriture)

print('proportion_cereales_animaux = ', proportion_cereales_animaux)
#print('proportion_cereales_nourriture = ',proportion_cereales_nourriture)


proportion_cereales_animaux =  45.91400584899829


====================================================<br>
Proportion des céréales dédiée animaux =  <b>45.91 </b><br>
====================================================




In [74]:
# bis

# liste_code_pays_sous_alimentation = df_sous_alimentation2013.loc[df_sous_alimentation2013['Valeur']>0,'Zone']


# # liste_code_pays_sous_alimentation = df_sous_alimentation2013[~df_sous_alimentation2013['Valeur'].isin([0,np.NaN])]['Code zone'].unique()
# df_bis3 = df_prinicpale[df_prinicpale['Zone'].isin(liste_code_pays_sous_alimentation.unique())]

# m2 = df_bis3.groupby('Produit').sum().reset_index().sort_values(by='Exportations - Quantité (kg/an)', ascending=False)[:20]['Produit']
# df_m2 = df_bis3.loc[df_bis3['Produit'].isin(produits_exports),:]
# df_m2 = df_m2.sort_values(by='Exportations - Quantité (kg/an)', ascending=False)[:200]
# df_m2 = df_m2[['Produit','Nourriture (kg/an)','Aliments pour animaux (kg/an)','Autres utilisations (non alimentaire) (kg/an)','Disponibilité intérieure (kg/an)']]

# df_group = df_m2.groupby('Produit').sum()


# df_group['Ratio autres utilisations/ Disponibilité intèrieur'] = \
# df_group['Autres utilisations (non alimentaire) (kg/an)'] / \
# df_group['Disponibilité intérieure (kg/an)']


# df_group['Ratio nourriture animale/ nourriture'] = \
# df_group['Aliments pour animaux (kg/an)'] / \
# (df_group['Nourriture (kg/an)'] + df_group['Aliments pour animaux (kg/an)'])



# (df_group['Nourriture (kg/an)','Aliments pour animaux (kg/an)'].sum(axis=1))

# print(df_bis3.shape)
# print(df_prinicpale.shape)

<b> Les pays en sous alimentation</b>

In [75]:
pays_sous_alimentation = df_sous_alimentation2013.loc[df_sous_alimentation2013['Valeur']>0,'Zone']

<b> DF principale des pays en sous alimentation </b>

In [76]:
df_p_sous_alimentation = df_prinicpale.loc[df_prinicpale['Zone'].isin(pays_sous_alimentation.unique()),:]

<b>les 15 produits les plus exportés par ce groupe de pays.</b>

In [77]:
df_produits = df_p_sous_alimentation.groupby('Produit').sum().reset_index()
df_produits = df_produits.sort_values(['Exportations - Quantité (kg/an)'], ascending=False ).head(15)
produits_exports = df_produits['Produit']

In [78]:
# Parmi les données des bilans alimentaires au niveau mondial, sélectionnez les 200 plus grandes 
# importations de ces produits (1 importation = une quantité d'un produit donné importée par un pays donné)
df_produits = df_prinicpale.loc[df_prinicpale['Produit'].isin(produits_exports), :]
df_produits = df_produits.sort_values('Importations - Quantité (kg/an)', ascending=False)[:200]
df_produits = df_produits[['Produit','Nourriture (kg/an)','Aliments pour animaux (kg/an)','Autres utilisations (non alimentaire) (kg/an)','Disponibilité intérieure (kg/an)']]

# Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits.
df_group_produit = df_produits.groupby('Produit').sum()
df_group_produit['Ratio1'] = df_group_produit['Autres utilisations (non alimentaire) (kg/an)']/df_group_produit['Disponibilité intérieure (kg/an)']
df_group_produit['Ratio2'] = df_group_produit['Aliments pour animaux (kg/an)']/(df_group_produit['Aliments pour animaux (kg/an)']+df_group_produit['Nourriture (kg/an)'])

 # Question 12 : Donnez les 3 produits qui ont la plus grande valeur pour chacun des 2 ratios.

In [79]:
liste_ratio1 = df_group_produit.nlargest(3,['Ratio1'],keep='all').index.tolist()
liste_ratio2 = df_group_produit.nlargest(3,['Ratio2'],keep='all').index.tolist()

print('liste_ratio1 = ',liste_ratio1)
print('liste_ratio2 = ',liste_ratio2)

liste_ratio1 =  ['Huile de Palme', 'Manioc', 'Maïs']
liste_ratio2 =  ['Maïs', 'Poissons Pelagiques', 'Soja']


<b>Top 3 des produits du ratio1 :</b>
<ol>    
    <li>Huile de palme </li>
    <li>Manioc </li>
    <li>Maïs</li>
</ol>    

<b>Top 3 des produits du ratio2 :</b>
<ol>
    <li>Maïs </li>
    <li>Poissons Pelagiques </li>
    <li>Soja </li>
</ol>


<b> Si on prends en considération 20 produits à la place de 15 </b> alors: <br>
<b>Top 3 des produits du ratio1 :</b>
<ol>
    <li>huile de soja </li>
    <li>huile de palme </li>
    <li>manioc </li>
</ol>    

<b>Top 3 des produits du ratio2 :</b>
<ol>
    <li>orge </li>
    <li>mais </li>
    <li>manioc </li>
</ol>

# Réponse 13 : Nombre de tonnes de céréales libérées si les USA diminuaient leur production de produits animaux de 10% 

In [80]:
df_cereal_USA = df_prinicpale_cereal[df_prinicpale_cereal['Zone']=="États-Unis d'Amérique"].groupby('Zone').sum().copy()

In [81]:
cereal_libere_usa = (df_cereal_USA['Aliments pour animaux (kg/an)'].iloc[0] * 0.1 ) /1000
print('cereal_libere_usa = ',cereal_libere_usa)

cereal_libere_usa =  14009600.0


<b> 14009600 </b> tonnes de cérales pourraient être libérées si les USA diminuaient leur production de produits animaux de 10%

# Réponse 14 

## Proportion de manioc produit par la Thaïlande qui est exportée 

In [82]:
df_thailande_manioc = df_prinicpale[(df_prinicpale['Zone']=='Thaïlande') & (df_prinicpale['Produit']=='Manioc')].copy()

In [83]:
proportion_thailande_manioc = df_thailande_manioc['Exportations - Quantité (kg/an)'].iloc[0]/df_thailande_manioc['Production (kg/an)'].iloc[0]*100
print("proportion_thailande_manioc = ",proportion_thailande_manioc,"%")

proportion_thailande_manioc =  83.41272991928014 %


proportion_thailande_manioc =  83.41272991928014 %

## Proportion de personnes en sous-nutrition 

In [84]:
df_sous_alimentation2013[df_sous_alimentation2013['Zone']=='Thaïlande']['Valeur'].iloc[0] *1000 * 100/ df_prinicpale[df_prinicpale['Zone']=='Thaïlande']['Population'].iloc[0]  

8.356836937219262

Proportion de personnes en sous-nutrition = <b> 0.08356836937219263
</b>

In [85]:
df_prinicpale.to_csv('data_sql/df_prinicpale.csv',index = False)

In [86]:
df_sous_alimentation2013.to_csv('data_sql/df_sous_alimentation.csv',index = False)

In [87]:
df_population.to_csv('data_sql/df_population2013.csv',index = False)